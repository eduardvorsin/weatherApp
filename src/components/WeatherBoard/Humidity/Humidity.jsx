import PropTypes from 'prop-types';
import PieChart from '../../UI/PieChart/PieChart';
import classes from './Humidity.module.css';

export default function Humidity({
  relativeHumidity,
  apparentTemperature,
  className,
  testId,
}) {
  return (
    <div
      className={`${classes.humidity} ${className}`}
      data-testid={testId}
    >
      <h3 className={classes.humidity__title}>
        Относительная влажность
      </h3>

      <div className={classes.humidity__content}>
        <PieChart
          className={classes['humidity__pie-chart']}
          value={relativeHumidity}
        />

        <p className={classes['humidity__apparent-temperature']}>
          Ощущаемая температура:
          {' '}
          {`${apparentTemperature}°`}
        </p>
      </div>
    </div>
  );
}

Humidity.defaultProps = {
  className: '',
  relativeHumidity: 0,
  apparentTemperature: 0,
  testId: '',
};

Humidity.propTypes = {
  className: PropTypes.string,
  relativeHumidity: PropTypes.number,
  apparentTemperature: PropTypes.number,
  testId: PropTypes.string,
};
