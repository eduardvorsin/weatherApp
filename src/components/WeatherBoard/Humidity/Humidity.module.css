.humidity {}

.humidity__content {
  display: flex;
  align-items: center;
  justify-content: center;
}

.humidity__title,
.humidity__apparent-temperature {
  margin: 0;
}

.humidity__title {
  text-align: center;
  margin-bottom: 25px;
  font-size: 18px;
  max-width: 320px;
  margin-left: auto;
  margin-right: auto;
}

.humidity__pie-chart {
  margin-right: 15px;
  font-size: 34px;
  flex: 0 1 200px;
  min-width: 150px;
  height: auto;
}

@media(min-width:577px) {
  .humidity__pie-chart {
    margin-right: 30px;
  }
}

@media(min-width:993px) {
  .humidity__pie-chart {
    font-size: 46px;
  }
}

.humidity__apparent-temperature {}
