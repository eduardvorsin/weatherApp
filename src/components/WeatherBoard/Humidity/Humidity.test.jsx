import { render, screen } from '@testing-library/react';
import Humidity from './Humidity';

describe('Humidity component tests', () => {
  it('renders correctly', () => {
    render(<Humidity
      className="test-class"
      relativeHumidity={80}
      apparentTemperature={-17}
    />);

    expect(screen.getByText(/-17Â°/)).toBeInTheDocument();
    expect(screen.getByText(/80/)).toBeInTheDocument();
  });

  it('renders without props', () => {
    render(<Humidity />);

    expect(screen.getAllByText(/0/)).toHaveLength(2);
  });

  it('the classname is correctly assigned to the element', () => {
    render(<Humidity
      className="test-class"
      testId="humidity"
    />);

    expect(screen.getByTestId('humidity')).toHaveClass('test-class');
  });

  it('snapshot with default data', () => {
    render(<Humidity
      testId="humidity"
    />);

    expect(screen.getByTestId('humidity')).toMatchSnapshot();
  });

  it('snapshot with mock text', () => {
    render(<Humidity
      className="test-class"
      relativeHumidity={80}
      apparentTemperature={-17}
      testId="humidity"
    />);

    expect(screen.getByTestId('humidity')).toMatchSnapshot();
  });
});
