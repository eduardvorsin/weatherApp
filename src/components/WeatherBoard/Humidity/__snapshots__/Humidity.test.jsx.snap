// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Humidity component tests snapshot with default data 1`] = `
<div
  class="humidity "
  data-testid="humidity"
>
  <h3
    class="humidity__title"
  >
    Относительная влажность
  </h3>
  <div
    class="humidity__content"
  >
    <div
      class="pie-chart humidity__pie-chart"
      data-testid=""
    >
      <svg
        class="pie-chart__icon"
        fill="none"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          class="pie-chart__circle"
          cx="255"
          cy="255"
          r="245.25"
        />
        <circle
          class="pie-chart__circle"
          cx="255"
          cy="255"
          r="245.25"
          style="--value: 0;"
        />
      </svg>
      <p
        class="pie-chart__text"
      >
        0
        %
      </p>
    </div>
    <p
      class="humidity__apparent-temperature"
    >
      Ощущаемая температура:
       
      0°
    </p>
  </div>
</div>
`;

exports[`Humidity component tests snapshot with mock text 1`] = `
<div
  class="humidity test-class"
  data-testid="humidity"
>
  <h3
    class="humidity__title"
  >
    Относительная влажность
  </h3>
  <div
    class="humidity__content"
  >
    <div
      class="pie-chart humidity__pie-chart"
      data-testid=""
    >
      <svg
        class="pie-chart__icon"
        fill="none"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          class="pie-chart__circle"
          cx="255"
          cy="255"
          r="245.25"
        />
        <circle
          class="pie-chart__circle"
          cx="255"
          cy="255"
          r="245.25"
          style="--value: 80;"
        />
      </svg>
      <p
        class="pie-chart__text"
      >
        80
        %
      </p>
    </div>
    <p
      class="humidity__apparent-temperature"
    >
      Ощущаемая температура:
       
      -17°
    </p>
  </div>
</div>
`;
